enumerable.rb

```diff
@@ -0,0 +1,87 @@
+# module Enumerable
+module Enumerable
+  def my_each
+    for item in self
+      yield(item)
+    end
+    return self if block_given?
+  end
+
+  def my_each_with_index
+    count = 0
+    self.my_each do |item|
+      yield(item, count)
+      count += 1
+    end
+    return self if block_given?
+  end
+
+  def my_select
+    array = []
+    for item in self
+      array.push(item) if yield(item) == true
+    end
+    return array if block_given?
+  end
+
+  def my_all?
+    count = 0
+    for item in self
+      count += 1 if yield(item) == true
+    end
+    count == self.length
+  end
+
+  def my_any?
+    count = 0
+    for item in self
+      count += 1 if yield(item) == true
+    end
+    return true if count > 0
+  end
```

Comment:
This method returns nil if it doesn't find a single item. It should also be able to return false